samtranslator/translator/translator.py

```diff
@@ -269,6 +274,83 @@ def _get_resources_to_iterate(
+                    resource["Connectors"], source_logical_id, "Connectors", is_resource_attribute=True
+                ).to_be_a_map()
+
+                # Go through each of the connectors that have been attached and create a Serverless Connector resource
+                for connector_logical_id, connector_dict in resource["Connectors"].items():
+
+                    # The logical id for the transformed connector will be <source_logical_id> + <embedded_connector_logical_id>
+                    connector_logical_id = source_logical_id + connector_logical_id
+
+                    # can't use sam_expect since this is neither a property nor a resource attribute
+                    if not isinstance(connector_dict, dict):
+                        raise InvalidResourceException(connector_logical_id, f"{connector_logical_id} should be a map.")
+
+                    # mutates the connector_dict to make it a connector resource
+                    get_generated_connector(source_logical_id, connector_logical_id, connector_dict, document_errors)
+
+                    connectors[connector_logical_id] = connector_dict
+
+                # delete connectors key, so it doesn't show up in the transformed template
+                del resource["Connectors"]
+            except InvalidResourceException as e:
+                document_errors.append(e)
+
+    # go through the list of generated connectors and add it to resources dict
+    for connector_logical_id, connector_dict in connectors.items():
+        if connector_logical_id in resources:
+            document_errors.append(
+                DuplicateLogicalIdException(
+                    connector_logical_id, connector_logical_id, resources[connector_logical_id]["Type"]
+                )
+            )
+        else:
+            resources[connector_logical_id] = connector_dict
+
+
+def get_generated_connector(
+    source_logical_id: str, connector_logical_id: str, connector: Dict[str, Any], document_errors: List[Any]
+) -> None:
+    """
+    Adds the type, source id and any ResourceReference properties to the connector
+
+    :param str source_logical_id: Logical id of the resource the connector is attached to
+    :param str connector_logical_id: Logical id of the connector
+    :param dict connector: the properties of the connector including the Destination, Permissions and SourceReference [Optional]
+    """
+    connector_class = SamConnector(logical_id="SamConnector")
+    connector["Type"] = connector_class.resource_type
+
+    # No need to raise an error for this instance as the error will be caught by the parser
+    if "Properties" in connector and isinstance(connector["Properties"], dict):
+        properties = connector["Properties"]
+        properties["Source"] = {"Id": source_logical_id}
+        if "SourceReference" in properties:
+            try:
+                sam_expect(properties["SourceReference"], connector_logical_id, "SourceReference").to_be_a_map()
```

Comment:
Not sure the error checking is worth it. It's not very thorough, and it'll be checked properly later anyway.